python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -279,6 +279,16 @@ def __init__(
         #  by the AMCS.
         self.previous_state = MotionState.PARKED
 
+        # Keep track of the parameters of the latest moveAz command issued so
+        # repetition of the same command can be avoided. This is necessary in
+        # case the dome repeatedly is instructed to move to the same position
+        # with velocity == 0.0 since that may introduce large oscillations. The
+        # position is set to the dome park position.
+        self.latest_moveAz_command = {
+            "position": 360.0 - DOME_AZIMUTH_OFFSET,
+            "velocity": 0.0,
+        }
```

Comment:
Please consider a dataclass or an instance of the move command topic data. That reduces the risk of mis-spelling field names.

Saving topic data is a bit simpler because there is no need to define the data class, and you can literally just say `self.latst_moveAz_command = data` in do_moveAz. Also it provides a timestamp, in case it's of interest. But it does add some irrelevant fields. If you use the topic data then you can initialize it in the `__init__` after calling `super().__init__` with `self.latst_moveAz_command = self.cmd_moveAz.DataType(position=..., velocity=...)`, which avoids the need to test for null. You can recognize "never set" by testing that private_sndStamp=0.

Reply:
I opted for a dataclass.